auth.py

```diff
@@ -158,8 +158,11 @@ def __init__(self, username, password=None, auth_url=None):
     def __call__(self, request):
         """Inject cookies into requests as they are made"""
         if self.cookie and self.digest:
-            request.headers.update({"Content-Type": "text/xml; charset=utf-8",
-                                    "Cookie" : self.cookie})
+            if 'Content-Type' in request.headers:
+                request.headers.update({"Cookie" : self.cookie})
+            else:
+                request.headers.update({"Content-Type": "text/xml; charset=utf-8",
+                                        "Cookie" : self.cookie})
```

Comment:
This makes sense, however the logic would probably be better like this:
```python
headers = {
    "Content-Type": request.headers.get("Content-Type") or 'text/xml; charset=utf-8",
    "Cookie": self.cookie
}
request.headers.update(headers)
```
The reason for this is that header names are case-insensitive. `Content-Type` could also be `Content-type`, `content-type` or `conTenT-tYpE`. `request.headers.get()` allows us to get the header value case-insensitively, and if it doesn't exist, we can set it to our desired value.